datadog_checks_dev/datadog_checks/dev/tooling/snmp_profile_validator/validate_profile.py

```diff
@@ -0,0 +1,135 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import json
+from os import listdir
+from os.path import isfile, join
+
+import jsonschema
+import yaml
+
+from ..commands.console import abort, echo_failure, echo_info
+from ..constants import get_root
+
+
+def validate_profile(file, directory, verbose):
+
+    profiles_list = find_profiles(file, directory)
+    contents = read_profile(profiles_list)
+    errors = validate_with_jsonschema(contents, verbose)
+    produce_errors(errors, verbose)
+
+
+class Profile:
+    def __init__(self):
+        self.file_path = ""
+        self.errors = []
+        self.valid = True
+        self.contents = ""
+
+    def __repr__(self):
+        return self.file_path
+
+
+def find_profiles(file, directory):
+    profiles_list = []
+    profile = Profile()
+    if file:
+        profile.file_path = file
+        profiles_list.append(profile)
+        return profiles_list
+
+    if directory:
+        profiles_list = get_all_profiles_from_dir(directory)
+        return profiles_list
+
+    profiles_path = join(get_root(), "snmp", "datadog_checks", "snmp", "data", "profiles")
+    profiles_list = get_all_profiles_from_dir(profiles_path)
+    return profiles_list
+
+
+def get_all_profiles_from_dir(directory):
+    profiles_list = []
+    profiles_path = directory
+    dir_contents = [file for file in listdir(profiles_path) if isfile(join(profiles_path, file))]
+    for file in dir_contents:
+        profile = Profile()
+        profile.file_path = join(profiles_path, file)
+        profiles_list.append(profile)
+    return profiles_list
+
+
+def read_profile(profiles_list):
+    read_profiles = []
+    for profile in profiles_list:
+        with open(profile.file_path) as f:
+            file_contents = yaml.safe_load(f.read())
+            profile.contents = file_contents
```

Comment:
🤏 have you considered moving these into the `Profile` class as a `loadFromFile` function? Something like 

```
def load(self, file_path):
  self.file_path = file_path
  with open(self.file_path) as f:
    self.content = yaml.safe_load(f.read())
```

and then here, or in a merged function with `get_all_profiles_from_dir` have:

``` 
for profile_path in files_in_directory:
  profile = Profile()
  profile.load(profile_path)
```